### PR TITLE
yewtube: init at 2.9.0, replace mps-youtube

### PR DESCRIPTION
**Pull request by fgaz**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fgaz** [comment]
cc @k0ral @odi (can't request a review for some reason). would you like to be added to the maintainer list for yewtube?

**Comment by odi** [comment]
> cc @k0ral @odi (can't request a review for some reason). 

maybe because I am not part of the NixOS github organization?

> would you like to be added to the maintainer list for yewtube?

no its not necessary for me to be on the maintainer list of yewtube

**Comment by k0ral** [comment]
> would you like to be added to the maintainer list for yewtube?

Gladly, thank you :) .

**Comment by fgaz** [comment]
> maybe because I am not part of the NixOS github organization?

That's probably it. I think you should both request to be added [here](https://discourse.nixos.org/t/join-the-package-maintainer-team/3751), so you can get pinged by ofborg

> > would you like to be added to the maintainer list for yewtube?
> 
> Gladly, thank you :) .

Great! I just added you

File: pkgs/development/python-modules/pyreadline/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    extension = "zip";
+    hash = "sha256-RTBZL8LoWyWxqfeWZEM9oJI3waJw5NeOpao6LHIp4tE=";
+  };
+
+  # https://github.com/pyreadline/pyreadline/issues/74
+  doCheck = false;
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyreadline/default.nix:15]: Missing pythonImportsCheck or is that also broken?

File: pkgs/applications/misc/yewtube/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  # https://github.com/pyreadline/pyreadline/issues/74
+  doCheck = false;
+
+  # before check create a directory and redirect XDG_CONFIG_HOME to it
+  preCheck = ''
+    mkdir -p check-phase
+    export XDG_CONFIG_HOME=$(pwd)/check-phase
+  '';
+
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/yewtube/default.nix:36]: pythonImportsCheck or is that also broken?

File: pkgs/applications/misc/yewtube/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    yt-dlp
+  ];
+
+  # https://github.com/pyreadline/pyreadline/issues/74
+  doCheck = false;
+
+  # before check create a directory and redirect XDG_CONFIG_HOME to it
+  preCheck = ''
+    mkdir -p check-phase
+    export XDG_CONFIG_HOME=$(pwd)/check-phase
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/yewtube/default.nix:34]:
```suggestion
    HOME=$(mktemp -d)
    export XDG_CONFIG_HOME=$HOME
```

**Review by SuperSandro2000 — COMMENTED** [review]
You forgot to aliases to add python-aliases.nix.

File: pkgs/development/python-modules/pyreadline/default.nix (hunk 1 above)
Comment (fgaz) [reply pkgs/development/python-modules/pyreadline/default.nix]: Looks like pyreadline is not needed at all, I removed it

https://github.com/iamtalhaasghar/yewtube/pull/105

File: pkgs/applications/misc/yewtube/default.nix (hunk 3 above)
Comment (fgaz) [reply pkgs/applications/misc/yewtube/default.nix]: I changed it to `export XDG_CONFIG_HOME=$(mktemp -d)`

**Comment by fgaz** [comment]
Ping

**Pull request merged**